sympy/external/importtools.py

```diff
@@ -157,7 +157,29 @@ def import_module(module, min_module_version=None, min_python_version=None,
         # NOTE: StrictVersion() is use here to make sure a comparison like
         # '1.11.2' < '1.6.1' doesn't fail. There is not a straight forward way
         # to create a unit test for this.
-        if StrictVersion(modversion) < StrictVersion(min_module_version):
+        for _ in range(1):
+            # strict version comparison can easily throw errors
+            version_check_failed = True  # Default value
+            try:
+                version_check_failed = (StrictVersion(modversion) <
+                                        StrictVersion(min_module_version))
+                break
+            except ValueError as e:  # e.g. StrictVersion('1.1.post1')
+                warnings.warn("Versions %s and %s are not comparable with "
+                              "StrictVersion()"
+                              % (modversion, min_module_version),
+                              UserWarning)
+
+            try:
+                from pkg_resources import parse_version
+                version_check_failed = (parse_version(modversion) <
+                                        parse_version(min_module_version))
+                break
+            except ImportError:  # could pkg_resources be missing?
+                warnings.warn("pkg_resources is not available.",
+                              UserWarning)
```

Comment:
setuptools is not a required dependency of sympy. No need to warn a user if it isn't there.

Reply:
Addressed this.  I don't remember why I put the warning there.